utilities/include/color.hpp

```diff
@@ -27,9 +27,10 @@ class color{
 
         //return 8 bit int values
         constexpr color get_int() const{
-            int R = (r > 1.0) ? 255 : std::max(0, static_cast<int>(255*r));
-            int B = (b > 1.0) ? 255 : std::max(0, static_cast<int>(255*b));
-            int G = (g > 1.0) ? 255 : std::max(0, static_cast<int>(255*g));
+            // std:max to clamp negative values to 0 
+            std::uint8_t R = static_cast<uint8_t>((r > 1.0) ? 255 : static_cast<int>(255*r));
```

Comment:
This double static cast doesn't make any sense. What you should do is, instead,
```cpp
std::uint8_t R = (r > 1.0) ? 255 : static_cast<std::uint8_t>(255*r);
```

Reply:
Also, do include the `<cstddef>` header because these integral types are not native types.